src/meadowrun/run_job_core.py

```diff
@@ -497,6 +520,51 @@ def __init__(self, process_state: ProcessState) -> None:
         self.process_state = process_state
 
 
+class _GetResultsUnordered(Protocol):
+    def __call__(
+        self, *, workers_done: Optional[asyncio.Event]
+    ) -> AsyncIterable[Tuple[int, ProcessState]]:
+        ...
+
+
+class TaskException(Exception):
+    pass
+
+
+@dataclasses.dataclass(frozen=True)
+class TaskResult(Generic[_T]):
```

Comment:
I think we should add this to src/meadowrun/__init__.py and the generated docs as it's part of the public API